\documentclass[12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{R/qtl2}
\subtitle{high-dimensional data \& multi-parent populations}
\author{\href{http://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{http://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[2pt]
\scriptsize {\lolit Slides:} \href{http://bit.ly/ctc_2017}{\tt \scriptsize
  \color{foreground} bit.ly/ctc\_2017}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \vfill \hfill \includegraphics[height=6mm]{Figs/cc-zero.png} \vspace*{-1cm}

  \note{These are slides for a talk that I will give at the CTC
    meeting in Memphis, TN, on 14 Jun 2017.

    Source: {\tt https://github.com/kbroman/Talk\_Rqtl2\_CTC2017} \\
    Slides: {\tt http://bit.ly/ctc\_2017}
}
} }


\begin{frame}[c]{17 years of R/qtl}

\only<1>{\figw{Figs/rqtl_lines_code.pdf}{1.0}}
\only<2>{\figw{Figs/rqtl_lines_code2.pdf}{1.0}}

\end{frame}



\begin{frame}[c]{}

\centerline{\Large Good things}

\vspace{4mm}

\onslide<2>{
\begin{itemize}
\lolit
  \item HMM code
  \item basics of the user interface
  \item diagnostics and data visualization
  \item quite comprehensive
  \item quite flexible
\end{itemize}
}

\end{frame}


\begin{frame}{}

\vspace*{16.7mm}

\centerline{\Large Bad things}

\end{frame}

\begin{frame}[c,fragile]{Stupidest code ever}

\begin{center}
\begin{minipage}[c]{9.3cm}
\begin{semiverbatim}
\lstset{basicstyle=\normalsize}
\begin{lstlisting}[linewidth=9.3cm]
n <- ncol(data)
temp <- rep(FALSE,n)
for(i in 1:n) {
  temp[i] <- all(data[2,1:i]=="")
  if(!temp[i]) break
}
if(!any(temp)) stop("...")
n.phe <- max((1:n)[temp])
\end{lstlisting}
\end{semiverbatim}
\end{minipage}
\end{center}

\vspace{3mm}

\hfill \href{https://kbroman.wordpress.com/2011/08/17/the-stupidest-r-code-ever}{\scriptsize \lolit \tt kbroman.wordpress.com/2011/08/17/the-stupidest-r-code-ever}

\end{frame}


\begin{frame}[c]{Input file}

\figh{Figs/datafile.pdf}{0.6}

\end{frame}


\begin{frame}[c]{}

  \large

  {\hilit Open source} {\lolit means}

  everyone can see my stupid mistakes

  \bigskip \bigskip \bigskip

  \onslide<2>{
    {\hilit Version control} {\lolit means}

  everyone can see every stupid mistake I've ever made
}

\end{frame}



\begin{frame}{}

\vspace*{16.7mm}

\centerline{\Large Support}

\end{frame}


\begin{frame}[c]{The ``future''}

\figh{Figs/mouse_on_chips.png}{0.75}

\end{frame}


\begin{frame}{Multi-parent populations}

\vspace{13mm}

\figw{Figs/valdar_genet2006.png}{1.0}

\small \color{myyellow}
\hspace*{19mm} combine \hspace*{15mm} mix \hspace*{23mm} fix

\vspace{17mm}

\hfill \href{https://doi.org/10.1534/genetics.104.039313}{\scriptsize \lolit Valdar et al., Genetics 172:1783, 2006}

\end{frame}


\begin{frame}{Challenges: {\color{foreground} diagnostics}}

\vspace{2mm}

\figw{Figs/weird_correlation_matrix.png}{0.65}

\vspace{3mm}

\hfill \href{https://kbroman.wordpress.com/2012/04/25/microarrays-suck}{\scriptsize \lolit \tt kbroman.wordpress.com/2012/04/25/microarrays-suck}

\end{frame}




\begin{frame}[c]{Challenges: {\color{foreground} scale of results}}

\only<1>{\figw{Figs/scale_fig1.pdf}{1.0}}
\only<2>{\figw{Figs/scale_fig2.pdf}{1.0}}

\end{frame}




\begin{frame}[c]{Challenges: {\color{foreground} organizing, automating}}

\only<1>{\figw{Figs/batches_fig1.pdf}{1.0}}
\only<2>{\figw{Figs/batches_fig2.pdf}{1.0}}
\only<3>{\figw{Figs/batches_fig3.pdf}{1.0}}
\only<4>{\figw{Figs/batches_fig4.pdf}{1.0}}
\only<5>{\figw{Figs/batches_fig5.pdf}{1.0}}
\only<6>{\figw{Figs/batches_fig6.pdf}{1.0}}
\only<7>{\figw{Figs/batches_fig7.pdf}{1.0}}

\end{frame}


\begin{frame}[c]{Challenges: {\color{foreground} metadata}}


  \centerline{What the heck is {\hilit \tt "FAD\_NAD SI 8.3\_3.3G"}?}

\end{frame}



\begin{frame}[c]{}
\centerline{\Large What was the question again?}
\end{frame}



\begin{frame}[c]{}

  \vspace*{5mm}

\figh{Figs/rqtl2_3d.png}{0.85}

\end{frame}



\begin{frame}[c]{R/qtl2}

\vspace*{-16.2mm}

  \vspace{21mm}

  \bbi
\item High-density genotypes
\item High-dimensional phenotypes
\item Multi-parent populations
\item Linear mixed models
  \ei

  \vspace{25mm}

\hfill \href{http://kbroman.org/qtl2}{\small \tt kbroman.org/qtl2}

\end{frame}



\begin{frame}[c]{R/qtl2: \color{foreground} Let's not make the same mistakes}

  \bbi
\only<1>{
\item C++ and Rcpp
\item Roxygen2 for documentation
\item Unit tests
\item A single ``switch'' for cross type
}
\only<2>{
{\lolit
\item C++ and Rcpp
\item Roxygen2 for documentation
\item Unit tests
\item A single ``switch'' for cross type
}
}
\onslide<2>{
\item Split into multiple packages
\item Yet another data input format
\item Flatter data structures, but still complex
}
\ei

\end{frame}



\begin{frame}[c]{R/qtl2 cross types}

  \bbi
\item backcross
\item intercross
\item 2-, 4-, 8-, and 16-way RIL by selfing
\item 2-, 4-, and 8-way RIL by sib-mating
\item 2-way AIL
\item Heterogeneous Stock (HS)
\item Diversity Outbreds (DO)
\item DO-F1
\item 19-way MAGIC
  \ei


\end{frame}



\begin{frame}[c]{Initial goal}

\begin{center}
Everything in \href{https://www.jax.org/research-and-faculty/faculty/research-scientists/daniel-gatti}{Dan Gatti}'s
\href{https://github.com/dmgatti/DOQTL}{DOQTL},  \\[8pt]
but for more general crosses
\end{center}

\end{frame}


\begin{frame}[c]{Genome scan}
  \figw{Figs/rqtl2_scan.pdf}{1.0}
\end{frame}

\begin{frame}[c]{LMMs}
  \figw{Figs/rqtl2_scan17.pdf}{1.0}
\end{frame}

\begin{frame}[c]{QTL effects}
  \figw{Figs/rqtl2_coef.pdf}{1.0}
\end{frame}

\begin{frame}[c]{BLUPs}
  \figw{Figs/rqtl2_blup.pdf}{1.0}
\end{frame}

\begin{frame}[c]{SNP associations}
  \figw{Figs/rqtl2_snpasso.pdf}{1.0}
\end{frame}



\begin{frame}[c]{Acknowledgments}

\begin{columns}[T]
  \begin{column}[T]{0.5\textwidth}
    \vspace{0pt}
\bi
\item[] Danny Arends
\item[] Gary Churchill
\item[] Nick Furlotte
\item[] Dan Gatti
\item[] Ritsert Jansen
\item[] Pjotr Prins
\item[] \'Saunak Sen
\item[] Petr Simecek
\item[] Artem Tarasov
\item[] Hao Wu
\item[] Brian Yandell
  \ei
  \end{column} \hfill
\begin{column}[T]{0.5\textwidth}
\vspace*{0mm}

  \bi
\item[] Robert Corty
\item[] Timoth\'ee Flutre
\item[] Lars Ronnegard
\item[] Rohan Shah
\item[] Laura Shannon
\item[] Quoc Tran
\item[] Aaron Wolen
\item[]
\item[] NIH/NIGMS
  \ei
\end{column}
\end{columns}

\end{frame}


\begin{frame}[c]{}

\Large

Slides: \href{http://bit.ly/ctc_2017}{\tt bit.ly/ctc\_2017} \quad
\includegraphics[height=5mm]{Figs/cc-zero.png}

\vspace{7mm}

\href{http://kbroman.org}{\tt \lolit kbroman.org}

\vspace{7mm}

\href{http://kbroman.org/qtl2}{\tt kbroman.org/qtl2}

\vspace{7mm}

\href{https://github.com/kbroman}{\tt \lolit github.com/kbroman}

\vspace{7mm}

\href{https://twitter.com/kwbroman}{\tt \lolit @kwbroman}


\end{frame}




\end{document}
